### PR TITLE
Js/build speed up

### PR DESCRIPTION
**Pull request by jakubswierczek**
[Story](https://app.shortcut.com/oazo-apps/story/4160/improve-deployment-process-10-15-minutes)

# Changes
* `Build` only with PR
* aws-staging - push only to AWS
* aws-staging, aws-pro - cache in registry
* docker hub - on demand with commit hash

### CODE REVIEW

File: .github/workflows/dockerhub.yaml [hunk 1]
```diff
@@ -1,53 +1,69 @@
...
-                     -t $DOCKERHUB_ORG/$REPO_NAME:$LATEST_TAG \
-                     .
-        docker push $DOCKERHUB_ORG/$REPO_NAME --all-tags
+      - name: Build, tag, and push image toDockerhub
+        id: build-image
+        uses: docker/build-push-action@v3
+        env:
+          SHA_TAG: ${{ github.event.inputs.commit_hash }}
+          LATEST_TAG: latest
+          ENVIRONMENT_TAG: staging
```
Comment (cristidas) [inline .github/workflows/dockerhub.yaml:41]: @jakubswierczek Is this tag still needed if the images from Docker Hub are not used on AWS?

File: .github/workflows/aws-staging.yaml [hunk 2]
```diff
@@ -30,57 +30,55 @@ jobs:
...
+
+
     - name: Build, tag, and push image to Amazon ECR and Dockerhub
       id: build-image
+      uses: docker/build-push-action@v3
       env:
         SHA_TAG: ${{ steps.vars.outputs.sha_short }}
         LATEST_TAG: latest
         ENVIRONMENT_TAG: staging
         ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
```
Comment (cristidas) [inline .github/workflows/aws-staging.yaml:53]: @jakubswierczek you also need to set `ECR_REPO_NAME` here

File: .github/workflows/aws-staging.yaml [hunk 3]
```diff
@@ -30,57 +30,55 @@ jobs:
...
+    - name: Set up Docker Buildx
+      id: buildex
+      uses: docker/setup-buildx-action@v2
+      with:
+        install: true
+
+
     - name: Build, tag, and push image to Amazon ECR and Dockerhub
       id: build-image
+      uses: docker/build-push-action@v3
```
Comment (cristidas) [inline .github/workflows/aws-staging.yaml:48]: @jakubswierczek have you tested this works with `aws-actions/amazon-ecr-login` ? IIRC, this Action needs `https://github.com/marketplace/actions/docker-login#aws-elastic-container-registry-ecr`

Comment (jakubswierczek) [reply .github/workflows/aws-staging.yaml:48]: oh, thanks for it. I forgot about it. I tested only with dockerhub.

File: .github/workflows/dockerhub.yaml (hunk 1 above)
Comment (jakubswierczek) [reply .github/workflows/dockerhub.yaml:41]: removed

**Comment by marcinciarka** [comment]
Closing old PR (cleaning up PR tab).

**Pull request closed without merging**